TypeCobol/Compiler/Scanner/Scanner.cs

```diff
@@ -154,17 +175,52 @@ public class Scanner
             }
 
             // Create a stateful line scanner, and iterate over the tokens
-            Scanner scanner = new Scanner(line, startIndex, lastIndex, tokensLine, compilerOptions, true, multiStringConcatBitPosition);
+            Scanner scanner = new Scanner(line, startIndex, lastIndex, tokensLine, compilerOptions, true, continuationPositions);
             Token nextToken = null;
-            while((nextToken = scanner.GetNextToken()) != null)
+            Token lastToken = null;
+            while ((nextToken = scanner.GetNextToken()) != null)
             {
                 if (nextToken.TokenType == TokenType.AlphanumericLiteral && (!nextToken.HasOpeningDelimiter || !nextToken.HasClosingDelimiter))
                 {
                     tokensLine.AddDiagnostic(MessageCode.SyntaxErrorInParser,
                         tokensLine.Indicator.StartIndex, tokensLine.Indicator.EndIndex, "Literal is not correctly delimited.");
                 }
+                bool nextTokenIncontinutaionLine = continuationPositions?.ContinuteBitPositions?.Get(nextToken.StartIndex) ?? false;
+                // Check if the previous AlphaNumericLiteral should be followed by a blank
+                bool previousAphaNumLitNeedSeparator = lastToken != null && lastToken.TokenType == TokenType.AlphanumericLiteral &&
+                    nextToken.TokenType != TokenType.SpaceSeparator && nextToken.TokenType != TokenType.PeriodSeparator &&
+                    ScannerUtils.IsIdentifierOrLiteral1(nextToken);
+                // Special check: not for a next token in a continuation line if the previous token was a literal string
+                if (previousAphaNumLitNeedSeparator)
+                {
+                    previousAphaNumLitNeedSeparator = !nextTokenIncontinutaionLine;
+                }
+
+                // Check if the next AlphaNumericLiteral should be preceed by a blank
+                bool nextAphaNumLitNeedSeparator = lastToken != null && lastToken.TokenType != TokenType.SpaceSeparator && lastToken.TokenType != TokenType.PeriodSeparator &&
+                    nextToken.TokenType == TokenType.AlphanumericLiteral &&
+                    ScannerUtils.IsIdentifierOrLiteral1(lastToken);
```

Comment:
typo: "nextAlphaNumLitNeedsSeparator"
this test can be factorized in a method so as to be used both at line 190 for `lastToken` and here for `nextToken`.

Reply:
Changed in 
c7580c1